Enable expression-literals for sqlite and starburst

appleby opened the pull request:
Closes QUE-855
Closes QUE-856

### Description

* Enable the `:expression-literals` feature for sqlite and starburst
* Rename True and False expressions in literal expression tests
* Update literal expression tests to work with drivers that treat empty string as null
* Add `empty-string-literal-expression-test` and remove empty string form `standard-literal-expression-defs`

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

appleby committed:
Rename True and False expressions in literal expression tests

Speculative fix for sqllite failing the tests.

appleby committed:
Update literal expression tests to work with drivers that treat empty string as null

appleby committed:
Add empty-string-literal-expression-test and remove empty string form standard-literal-expression-defs

appleby committed:
Fix logical merge conflicts in case-with-literal-expression-test

appleby committed:
Enable the :expression-literals feature for sqlite and starburst

metamben commented on test/metabase/query_processor/test_util.clj:
```diff
@@ -276,6 +276,19 @@
     (1 1M true)  true
     (throw (ex-info "value is not boolish" {:value x}))))
 
+(defn format-nil->empty-str
+  "Return a function that converts nil to an empty string if driver treats empty strings a null.
+
+  Can be used with [[format-rows-by]] to normalize empty strings in results (but don't forget you need to pass
+  `format-nil-values?` when calling the [[format-rows-by]])."
+  [driver]
+  (if-not (driver.tu/empty-string-is-null? driver)
+    str
+    (fn nil->empty-string [value]
+      (if (nil? value)
+        ""
+        (str value)))))
```
Since
```clojure
user> (str nil)
""
```
this is could be simplified to `(def format-nil->empty-str str)`, so I wouldn't even bother introducing it.

appleby replied:
Aha! I mistakenly assumed `(str nil)` was returning `nil` because I didn't know about the `format-nil-values?` arg to `format-rows-by`. Once I discovered it, it didn't occur to me that `format-nil-values?` was the whole explanation, and this function func is unnecessary. Good catch!

appleby committed:
Review feedback: drop format-nil->empty-str

As pointed out by metamben in code review `(str nil)` is already the empty string. In fact, the culprit here was a
missing `format-nil-values?` arg to `format-rows-by`.
